lib/index.js

```diff
@@ -409,6 +409,12 @@ Mongoose.prototype.connect = function(uri, options, callback) {
   const _mongoose = this instanceof Mongoose ? this : mongoose;
   const conn = _mongoose.connection;
 
+  if (_mongoose.options.strictQuery === undefined) {
+    utils.warn('Mongoose: the `strictQuery` option will be switched back to `false` by default ' +
```

Comment:
Let's add a `lib/helpers/printStrictQueryWarning.js` file, and make that file export a function that is wrapped in `[`util.deprecate()`](https://nodejs.org/api/util.html#utildeprecatefn-msg-code). Instead of manually printing using `util.warn()`. That will tie better into existing workflows for tracking deprecation warnings, and also make it so this warning isn't printed multiple times in tests.